src/refiners/foundationals/latent_diffusion/stable_diffusion_xl/text_encoder.py

```diff
@@ -40,21 +40,22 @@ def __init__(
     def init_context(self) -> Contexts:
         return {"text_encoder_pooling": {"end_of_text_index": []}}
 
-    def __call__(self, text: str) -> tuple[Float[Tensor, "1 77 1280"], Float[Tensor, "1 1280"]]:
+    def __call__(self, text: str) -> tuple[Float[Tensor, "batch 77 1280"], Float[Tensor, "batch 1280"]]:
         return super().__call__(text)
 
     @property
     def tokenizer(self) -> CLIPTokenizer:
         return self.ensure_find(CLIPTokenizer)
 
     def set_end_of_text_index(self, end_of_text_index: list[int], tokens: Tensor) -> None:
-        position = (tokens == self.tokenizer.end_of_text_token_id).nonzero(as_tuple=True)[1].item()
-        end_of_text_index.append(cast(int, position))
+        for str_tokens in tokens.split(1):  # type: ignore
```

Comment:
Let's add `split` to `torch` imports and replace with:

```python
for str_tokens in split(tokens, 1):
    ...
```
That way you can get rid of the `# type: ignore`